### PR TITLE
d8 port of #180- Use the payment processor payment method instead of the default c…

### PR DESCRIPTION
**Pull request by jitendrapurohit**
…onfigured one

Overview
----------------------------------------
d8 port for #180  

Before
----------------------------------------
Default payment instrument is set for the contribution.

After
----------------------------------------
Payment processor instrument is set for the contirbution done via webforms.

### CODE REVIEW

File: src/WebformCivicrmPostProcess.php [hunk 1]
```diff
@@ -2148,6 +2141,9 @@ private function contributionParams() {
         'domain_id' => $domain,
       ));
     }
+    if (empty($params['payment_instrument_id']) && !empty($params['payment_processor_id'])) {
```
Comment (jitendrapurohit) [inline src/WebformCivicrmPostProcess.php:2144]: included the changes from https://github.com/colemanw/webform_civicrm/pull/289 which was a fix for the regression caused by #180

Comment (KarinG) [reply src/WebformCivicrmPostProcess.php:2144]: @jitendrapurohit - can you please add a few lines to the `ContributionIatsTest` -> section line 180-186 to check that with this PR the `payment_instrument_id` is indeed no longer `check` (default) but `credit card`?

Comment (jitendrapurohit) [reply src/WebformCivicrmPostProcess.php:2144]: Done. Thanks for the pointer 👍  @KarinG

Comment (KarinG) [reply src/WebformCivicrmPostProcess.php:2144]: Awesome - it's just a few lines but nice to secure this 👍

**Comment by KarinG** [comment]
Squashing and merging ✅

**Pull request merged**